src/cloud-cli/meltano/cloud/api/config.py

```diff
@@ -102,6 +107,84 @@ def config_path(self) -> Path:
 
         return self._config_path
 
+    @staticmethod
+    def decode_jwt(token: str) -> dict[str, t.Any]:
+        """Decode a JWT without verifying.
+
+        Args:
+            token: the jwt to decode
+
+        Returns:
+            The decoded jwt.
+        """
+        return jwt.decode(token, options={"verify_signature": False})
+
+    @property
+    def trks_and_pids(self) -> t.List[str]:
+        """Get tenant resource keys and project ids from id token.
+
+        Returns:
+            List of trks and pids in the form '<trk>::<jpid>`
+
+        """
+        with suppress(jwt.DecodeError):
+            decoded = self.decode_jwt(self.id_token)
+            trks_and_pids = decoded.get("custom:trk_and_pid")
+            if trks_and_pids:
+                return [perm.strip() for perm in trks_and_pids.split(",")]
+        return []
+
+    @property
+    def tenant_resource_keys(self) -> t.List[str]:
+        """Get list of tenant resource keys from id token.
+
+        Returns:
+            List of trks in id token.
+
+        """
+        return [perm.split("::")[0] for perm in self.trks_and_pids]
+
+    @property
+    def internal_project_ids(self) -> t.List[str]:
+        """Get list of project IDs from id token.
+
+        Returns:
+            List of pids in id token.
+
+        """
+        return [perm.split("::")[1] for perm in self.trks_and_pids]
+
+    @property
+    def internal_project_id(self) -> str:
+        """Get the internal project ID.
+
+        Returns:
+            Internal project ID.
+
+        Raises:
+            NoMeltanoCloudProjectIDError: when ID token includes no project IDs.
+        """
+        try:
+            return self.internal_project_ids[0]
+        except IndexError:
+            raise NoMeltanoCloudProjectIDError()
+
+    @property
+    def tenant_resource_key(self) -> str:
+        """Get the tenant resource key.
+
+        Returns:
+            The tenant resource key.
+
+        Raises:
+            NoMeltanoCloudTenantResourceKeyError: when ID token includes no
+                tenant resource keys.
+        """
+        try:
+            return self.tenant_resource_keys[0]
+        except IndexError:
+            raise NoMeltanoCloudTenantResourceKeyError()
```

Comment:
These should raise an exception if there is ambiguity about which ID/key to use, rather than taking the first one. They should only return a value if there is exactly one in the collection.